### PR TITLE
Save and let Biglearn know which calculation and ecosystem matrix uuids we used

### PR DESCRIPTION
**Pull request by Dantemss**
Research data etc

- Receive calculation and ecosystem matrix uuids from biglearn-api
- Save used calculation and ecosystem matrix uuids in tasks
- Send used calculation and ecosystem matrix uuids back to biglearn-api

### CODE REVIEW

File: app/representers/api/v1/book_toc_representer.rb [hunk 1]
```diff
@@ -90,10 +90,6 @@ class BookTocRepresenter < Roar::Decorator
              }
 
     property :chapter_section,
-             getter: ->(*) {
-               baked_chapter_section.blank? ?
-                 chapter_section : baked_chapter_section
-             },
```
Comment (reedstrm) [inline app/representers/api/v1/book_toc_representer.rb]: Seems unrelated to the rest of the PR?

File: lib/openstax/biglearn/api/real_client.rb [hunk 2]
```diff
@@ -396,6 +396,19 @@ def create_update_assignments(requests)
...
+      spe_ecosystem_matrix_uuid = task.spe_ecosystem_matrix_uuid
+
+      pes = {}
+      pes[:calculation_uuid] = pe_calculation_uuid if pe_calculation_uuid.present?
+      pes[:ecosystem_matrix_uuid] = pe_ecosystem_matrix_uuid if pe_ecosystem_matrix_uuid.present?
+
+      spes = {}
+      spes[:calculation_uuid] = spe_calculation_uuid if spe_calculation_uuid.present?
+      spes[:ecosystem_matrix_uuid] = spe_ecosystem_matrix_uuid if spe_ecosystem_matrix_uuid.present?
+
```
Comment (reedstrm) [inline lib/openstax/biglearn/api/real_client.rb]: I assume these have to be guarded because old task-records won't have them?

File: app/representers/api/v1/book_toc_representer.rb (hunk 1 above)
Comment (nathanstitt) [reply app/representers/api/v1/book_toc_representer.rb]: yeah, it's duplicated with another `getter` below that will override this one, causing it to never be called

File: lib/openstax/biglearn/api/real_client.rb (hunk 2 above)
Comment (Dantemss) [reply lib/openstax/biglearn/api/real_client.rb]: Both tasks that never had PEs and/or SPEs assigned and old records will have `nil`s here, so I decided to send a {} instead of sending the nulls...
